Add a script for converting WRL to OBJ

csauve opened the pull request:
The WRL files produced by tool are a useful debugging tool but can't be imported by Blender. I've added a simple text-transforming script to reformat the line segments contained in the WRL as OBJ lines. These appear as floating edges when imported to Blender. Usage described in included README.md.

### Limitations
It's a pretty quick hack job, so it doesn't cover cases where the points list spans multiple lines (a case which I've just noticed after writing this) and doesn't construct faces from `IndexedFaceSet`, so this is mostly useful for open edges and other local degenerate cases. Errors are not highlighted by the colour either.

csauve committed:
Add a script for converting WRL to OBJ

ShellyHerself commented:
So, first off, I really appreciate the contribution.

But have you tried enabling VRML import in Blender in the settings? It seems to be an optional addon that comes shipped by default.

Or is this an issue with Blender's importer being 2.0 and not supporting 1.0?
![Screenshot_20200303_145723](https://user-images.githubusercontent.com/24233409/75782569-60786280-5d5f-11ea-8022-d2dfabdcabf7.png)

ShellyHerself commented:
Okay yeah, that's weird it just silently doesn't import Halo wrl files with that addon. That's stupid.

Could you add your author info somewhere?

ShellyHerself commented on blender_scripts/README.md:
```diff
@@ -0,0 +1,18 @@
+
+When importing the `.obj` into Blender, be sure to use Y forward and Z up settings. You can then easily see the errors by highlighting the imported object's edges in wireframe view:
+
+![](wrl-blender.png)
```
Maybe we could move the images to a resources folder. Like `rsrc/` or something so they won't clutter up the scripts folder.

ShellyHerself commented on blender_scripts/wrl-to-obj.py:
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/python
+import re
```
Could you add your author info as a comment here maybe?

csauve committed:
Rewrite WRL to OBJ converter script; move its readme to READMES dir

csauve commented:
I've implemented the change requests and rewritten the parser to not be a regex hack job. The script contains a grammar and what I've learned about the WRL format.

ShellyHerself commented on blender_scripts/wrl-to-obj.py:
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python
```
The MEK standard has been to signal for python3. Tatsu is also Python 3 only. We should probably tell it to launch that.

ShellyHerself reviewed (changes requested):
Impressive!
![Screenshot_20200306_005525](https://user-images.githubusercontent.com/24233409/76037773-303fe800-5f48-11ea-8d13-976d58e0ece1.png)

I would like it if we could make more of the script conform to 80 chars per line though.

This probably also shouldn't be in the blender_scripts folder as it's not really a blender script. Probably the tools_misc folder. But, to be honest we kind of need to redo all the scripts folders as they're getting a bit insane anyway.

I'll refer to @Mimickal for the next review. See what she thinks.

csauve committed:
Move WRL converter to tools_misc, update its docs and comments, prefer python3

ShellyHerself commented on tools_misc/wrl_to_obj.py:
```diff
@@ -0,0 +1,148 @@
+# The caller is responsible for closing streams.
+def convert_wrl_to_obj(input_stream, output_stream):
+    def output(line):
+        print(line, file=output_stream)
```
Python allows for documentation to stored in the first statement of a function/object definition.

Done as follows:
```py
def parse_wrl_to_ast(wrl_content):
    '''
    Given a string of the WRL content, returns its abstract syntax tree.
    '''
    return WRL_PARSER.parse(wrl_content)
```

This makes it so that whenever automatic documentation generators are ran over projects that function description actually ends up in the documentation.

ShellyHerself reviewed (changes requested):
I am very pleased. Thanks for being nice about the feedback and following up with changes. :smile: 

I have just one more thing I personally want before I'm ready to approve this.

csauve committed:
Add doc comments to WRL converter functions

ShellyHerself reviewed (approved):
This gets my approval. I like it :fox_face:

Mimickal reviewed (approved):
This is solid :+1:
